ruby_programming/advanced_ruby/blocks.md

```diff
@@ -287,30 +295,40 @@ a_lambda.call
 # => 1
 ~~~
 
-A Proc object returns from the context in which it is called. If you are in the top level context (outside of a class and method) then you'll get an error because you can't return out of the very top level context, as there is no caller to return to.
+In non-lambda procs, `return` means exit from embracing method in which it was *defined* and will throw `LocalJumpError` if invoked outside the method. Check the [documentation](https://ruby-doc.org/core-2.7.0/Proc.html#class-Proc-label-Lambda+and+non-lambda+semantics) about it. In the top level context (not inside any method) a `return` will exit the program.
```

Comment:
What do you mean in the top-level context a return will exit the program?

Reply:
In top-level-context a return will have the same effect as 'exit 0'.

Is that better, or is there some other thing unclear about it?

Reply:
That's not something students will be very clear about